web_widget_image_webcam/static/src/js/webcam_widget.js

```diff
@@ -9,129 +9,124 @@ odoo.define("web_widget_image_webcam.webcam_widget", function (require) {
     var core = require("web.core");
     var rpc = require("web.rpc");
     var Dialog = require("web.Dialog");
-    var FieldBinaryImage = require("web.basic_fields").FieldBinaryImage;
+    const {patch} = require("@web/core/utils/patch");
+    const {ImageField} = require("@web/views/fields/image/image_field");
 
     var _t = core._t;
     var QWeb = core.qweb;
 
-    FieldBinaryImage.include({
-        _render: function () {
-            this._super();
-
-            var self = this,
-                WebCamDialog = $(QWeb.render("WebCamDialog")),
-                img_data = false;
+    patch(ImageField.prototype, "web_widget_image_webcam", {
+        setup() {
+            var self = this;
+            self._super.apply(self, arguments);
+            self.flash_fallback_mode = false;
+            rpc.query({
+                model: "ir.config_parameter",
+                method: "get_webcam_flash_fallback_mode_config",
+            }).then(function (default_flash_fallback_mode) {
+                self.flash_fallback_mode = default_flash_fallback_mode;
+            });
+        },
 
+        _setWebcamParams() {
             // ::webcamjs:: < https://github.com/jhuckaby/webcamjs >
             // Webcam: Set Custom Parameters
+            var self = this;
```

Comment:
This is totally unnecessary.
The `var self = this` trick is old, and useless in modern js. It was used to keep a reference for `this` within callback functions defined within the method.. so it's even not needed here, as there aren't any callbacks in this method.

Simply read `flash_fallback_mode` from `this`

Reply:
sure thanks! I'll fix that